### ISSUE TITLE
[Tune] Incorrectly schedule tasks that exceed the available resource

### ISSUE THREAD

**Issue by IcarusWizard** [ray-project/ray#17479]
### What is the problem?
I was running a hyper-parameter search for `pytorch` models with `tune`. The resource requirement of each task is set as `resources_per_trial={"cpu": 1, "gpu": 0.33,}`. Everything goes well in the first round, there are three tasks running since I have 1 GPU.  However, in the second round there are four tasks running at the same time. From the terminal, I see
`Resources requested: 4/8 CPUs, 1.32/1 GPUs, 0.0/74.41 GiB heap, 0.0/24.76 GiB objects (0/1.0 accelerator_type:V100)`. And the extra task cannot access to GPU (by checking `torch.cuda.is_available()`).

*Ray version and other system information (Python version, TensorFlow version, OS):*
Ubuntu
python=3.7
ray=1.2

### Reproduction (REQUIRED)
I cannot reproduce the same problem with a simple demo. I guess the problem is with the timing of release resources from the finished tasks and schedule new tasks. My model is a bit heavy which may take a longer time to release. 

- [x] I have verified my script runs in a clean environment and reproduces the issue.
- [ ] I have verified the issue also occurs with the [latest wheels](https://docs.ray.io/en/master/installation.html).

**Comment by Yard1** [comment]
I do not believe this is a bug. The scheduler may queue trials that are over the resource limit, and they will simply wait until resources become free. As long as the other trials are not stuck in pending but running as expected, this is normal behavior. What do you think @richardliaw ?

**Comment by Yard1** [comment]
CC @xwjiang2010

**Comment by richardliaw** [comment]
Yeah, we should raise an error when this is happening :)

**Comment by xwjiang2010** [comment]
@IcarusWizard Do you mind checking if `queue_trials` is incorrectly set to `True` in your case? This is used to trigger autoscaling. Your case looks like a single node; so this field should be set to `False`.

**Comment by IcarusWizard** [comment]
> @IcarusWizard Do you mind checking if `queue_trials` is incorrectly set to `True` in your case? This is used to trigger autoscaling. Your case looks like a single node; so this field should be set to `False`.

Just check the code. Yes, I have set `queue_trails` to `True`. But why it would cause the problem?

**Comment by Yard1** [comment]
`queue_trials=True` will purposefully schedule more trials than can be run currently so that the Ray autoscaler feels pressure to add more nodes. If your cluster is not expected to autoscale, `queue_trials` should be set to its default value of False.

**Comment by xwjiang2010** [comment]
`queue_trials` is deprecated now. Closing out the bug.
